hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -295,7 +295,7 @@ public void close() {
     try {
       ddlExecutor.close();
       if (client != null) {
-        client.close();
+        Hive.closeCurrent();
         client = null;
```

Comment:
What's the difference ?

Reply:
We get a new hms client as follows 

`Hive.get(configuration).getMSC(); `

which creates the client and saves it in a ThreadLocal map as well.
So if we ask for hms client at multiple places then close can only be called on the client only once.
So we resort to 

`Hive.closeCurrent()`

 which first checks in the ThreadLocal map and if the client is there, then only tries to close it. So further invocations don't throw exceptions

Reply:
Thanks for the explanation.

Reply:
@jsbali I just quickly search the whole project, it seems there are still other cases that call the `close` method directly. Can you fix them passingly?

Reply:
Sure @yanghua will take a look and fix those as well

Reply:
@jsbali do you plan to update this PR with the other fixes as well?